File: src/main/java/codereview/simpleorder/item/repository/ItemRepository.java
package codereview.simpleorder.item.repository;

import codereview.simpleorder.item.domain.Item;
import jakarta.persistence.LockModeType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Lock;

import java.util.Collection;
import java.util.List;

public interface ItemRepository extends JpaRepository<Item, Long> {

    @Lock(value = LockModeType.OPTIMISTIC)
    List<Item> findByIdIn(Collection<Long> itemIds);

    List<Item> findAll();
}

File: item/src/main/kotlin/msa/with/ddd/item/repository/ItemRepository.kt
package msa.with.ddd.item.repository

import msa.with.ddd.item.domain.entity.Item
import org.springframework.data.jpa.repository.JpaRepository

interface ItemRepository : JpaRepository<Item, Long> {
    
    fun findByIdIn(itemIds: Collection<Long>): List<Item>
}